### PR TITLE
Allow toggling IFA tracking for order reporting and post install requests

### PR DESCRIPTION
**Pull request by kirtanp98**
### Goals :dart:
Merchants can toggle IFA when reporting an order. Additionally, merchants can toggle IFA for post install requests.

### Implementation Details :construction:
* `ReportOrderBody` now has the `advertisingId` property.
* `PostInstallBody`'s `ifa` property is now optional and is set by `advertisingId`.
* `PostInstallBody`'s `ifaLimited` property is removed.
*  Made `adIdManager` a private property in `System`.
* `advertisingId` only get set when `includesIFA` and `adIdManager.isAdvertisingTrackingEnabled` is true, otherwise it is nil.

### Testing Details :mag:
Updated and addded some unit tests.

### CODE REVIEW

File: Tests/UnitTests/SystemTests.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ class SystemTests: XCTestCase {
                              bundle: bundle)
         XCTAssertEqualReferences(system.fileManager, fileManager)
         XCTAssertEqualReferences(system.calendar as AnyObject, calendar)
-        XCTAssertEqualReferences(system.adIdManager, adIdManager)
+        //XCTAssertEqualReferences(system.adIdManager, adIdManager)
```
Comment (wessmith) [inline Tests/UnitTests/SystemTests.swift]: You left some commented code here

File: Tests/UnitTests/TestObjects/TestSystem.swift [hunk 2]
```diff
@@ -36,12 +38,19 @@ final class TestSystem: SystemType {
...
     var fileManager: FileManagerType
     var calendar: CalendarType
-    var adIdManager: ASIdentifierManagerType
     var device: UIDeviceType
     var screen: UIScreenType
     var locale: LocaleType
     var bundle: BundleType
-
+    
+    var advertisingId: String? {
```
Comment (wessmith) [inline Tests/UnitTests/TestObjects/TestSystem.swift]: best not to reimplement the actual logic here. You could add a property to the test class here to configure what you want this to return for the class you're testing.

File: Tests/UnitTests/ReportOrderBodyTests.swift [hunk 3]
```diff
@@ -71,7 +71,8 @@ class ReportOrderBodyTests: XCTestCase {
         
         // Assert
         XCTAssertEqual(body.dictionaryRepresentation as NSDictionary,
-                       ["btn_ref": "srctok-abc123",
+                       ["advertising_id": "00000000-0000-0000-0000-000000000000",
```
Comment (wessmith) [inline Tests/UnitTests/ReportOrderBodyTests.swift]: I'd like to see a test for when the `advertising_id` is disabled asserting that the payload omits the field.

File: Source/System.swift [hunk 4]
```diff
@@ -27,7 +27,7 @@ import UIKit
 internal protocol SystemType: Configurable {
     var fileManager: FileManagerType { get }
     var calendar: CalendarType { get }
-    var adIdManager: ASIdentifierManagerType { get }
```
Comment (danielx0328) [inline Source/System.swift]: why was this removed?

File: Source/System.swift [hunk 5]
```diff
@@ -45,15 +45,24 @@ internal protocol SystemType: Configurable {
 
 internal final class System: SystemType {
     
+    private var adIdManager: ASIdentifierManagerType
```
Comment (danielx0328) [inline Source/System.swift]: Are there any issues with mocking a `System` for tests now that `SystemType` does not declare `adIdManager`?

File: Source/System.swift (hunk 4 above)
Comment (kirtanp98) [reply Source/System.swift]: We want `adIdManager` to be only accessed by the `System` class, so I had to move the property into the `System` class because the `SystemType` protocol wouldn't allow for a private property.

File: Source/System.swift (hunk 5 above)
Comment (kirtanp98) [reply Source/System.swift]: No issues, I just needed to include the `adIdManager` property to `TestSystem`

File: Source/System.swift (hunk 4 above)
Comment (wessmith) [reply Source/System.swift]: @danielx0328 The ifa is now optional in requests depending on user & brand configuration (e.g. user disabled in Settings, brand sets to false). Given that, it makes sense to enforce proper usage internally in the library. We don't want to expose access to the adIdManager directly or we might inadvertently attach an ifa without checking the two above conditions. The change here reflects that—as the system now exposes the `advertisingId` directly as an optional and returns a value based on those conditions. All consumers must use the correct property on System and the conditions are guaranteed to considered before returning the value.

File: Source/System.swift (hunk 5 above)
Comment (wessmith) [reply Source/System.swift]: Note that the adIdManager is still passed into the System constructor—so the output on the system instance `advertisingId` is still constrolled by the injected adIdManager.
